src/google/protobuf/compiler/php/php_generator.cc

```diff
@@ -663,6 +688,16 @@ void GenerateEnumToPool(const EnumDescriptor* en, io::Printer* printer) {
   Outdent(printer);
 }
 
+void GenerateServiceMethod(const MethodDescriptor* method,
+                           io::Printer* printer) {
+  printer->Print(
+        "public function ^camel_name^(^request_name^ $request);\n\n",
+        "camel_name", UnderscoresToCamelCase(method->name(), false),
+        "request_name", UnderscoresToCamelCase(
+          method->input_type()->name(), true)
```

Comment:
Good point.  I was able to break it by defining the request in a separate proto, importing it, and setting different namespaces on each.